apps/os/backend/agent/context.ts

```diff
@@ -106,6 +107,101 @@ export const defineRule = <Rule extends ContextRule>(rule: Rule) => rule;
 
 export const defineRules = <Rules extends ContextRule[]>(rules: Rules) => rules;
 
+/**
+ * Parse YAML front matter from a markdown string and return { attrs, body }.
+ * If no front matter is present, attrs is {} and body is the original string.
```

Comment:
let's mimic the format of the most popular library that does this: https://www.npmjs.com/package/gray-matter

so return type would be `{ content: string; data: Record<string, unknown> }`

unless this is based on something else that's already standard-ish?